### PR TITLE
Support schema settings

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
Schemas did not support settings, but it was needed for PublicKey schemas

Solution
========
- Added support for schema settings

Steps to Verify:
----------------
1. Deploy publicKey
2. Check the Chain state for schemas and see that settings for the new schema is set to 3.

### CODE REVIEW

**Review by saraswatpuneet — APPROVED** [review]
lgtm, especially adding the new settings

**Review by shannonwells — APPROVED** [review]
Awesome!

File: deploy.ts [hunk 1]
```diff
@@ -58,12 +81,46 @@ const createSchemas = async (schema_names: string[]) => {
...
+                console.log("SUCCESS: " + schemaName + " schema proposed with id of " + id + " and hash of " + hash);
+              }
+              resolve();
+            }
+          });
+      });
+      promises.push(promise);
+    } else {
+      // Create directly via sudo
+      const tx = api.tx.schemas.createSchemaViaGovernance(
```
Comment (aramikm) [inline deploy.ts:113]: What if it already exists? Do we need a way to dedup schemas before uploading?

Comment (wilwade) [reply deploy.ts:113]: We don't yet in this utility. There is a `read` tool that will figure it out, but deploy does not check it yet. Not really sure it should unless it is mainnet, but deploying to mainnet is always scary.

File: deploy.ts [hunk 2]
```diff
@@ -58,12 +81,46 @@ const createSchemas = async (schema_names: string[]) => {
     // Remove whitespace in the JSON
     const json_no_ws = JSON.stringify(JSON.parse(json));
 
-    const promise = new Promise<void>((resolve, reject) => {
-      api.tx.schemas
-        .createSchema(json_no_ws, schemaDeploy.modelType, schemaDeploy.payloadLocation)
-        .signAndSend(signerAccountKeys, { nonce: nonce++ }, ({ status, events, dispatchError }) => {
+    if (shouldPropose) {
+      // Propose to create
+      const promise = new Promise<void>((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (wilwade) [reply deploy.ts:86]: Good idea. I'll see if we could clean that up in the next update.

**Pull request merged**